src/OpenTelemetry/Metrics/MetricReaderTemporalityPreference.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="MetricReaderTemporalityPreference.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Metrics;
+
+/// <summary>
+/// Defines the behavior of a <see cref="MetricReader" />
+/// with respect to <see cref="AggregationTemporality" />.
+/// </summary>
+public enum MetricReaderTemporalityPreference
+{
+    /// <summary>
+    /// All aggregations are performed using cumulative temporatlity.
+    /// </summary>
+    Cumulative = 1,
+
+    /// <summary>
+    /// All measurements that are monotnic in nature are aggregated using delta temporality.
+    /// Aggregations of non-monotonic measurements use cumulative temporality.
+    /// </summary>
+    Delta = 2,
```

Comment:
@alanwest Thinking out loud here, could we be more descriptive with the names? Like...

```csharp
AlwaysCumulative,
DeltaMonotonicCumulativeNonmonotonic
```

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md The OTLP ENV variable called this simply cumulative, delta. 
Though I agree a more descriptive name is nicer, I also prefer if we keep it close to the spec. (its really an OTLP spec, but thats the only place spec ever mentions this...)

Reply:
They are totally waaay more descriptive 😆. I originally had them as `Cumulative` and `MonotonicDelta`, but the spec has no such term. In discussing with @cijothomas, we decided to keep the names from the OTLP spec used for the `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` environment variable - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md. We have not yet implemented this environment variable, but the meaning of the values `Cumulative` and `Delta` will map to this new enum.

Reply:
Oops sorry my response was redundant... Is it just me or does GitHub not seem to refresh stuff reliably?

Reply:
> Is it just me or does GitHub not seem to refresh stuff reliably?

Same here, I keep seeing such issue.

Reply:
I personally don't have an issue with the environment variable values being different from the enumeration names, but if that's what you guys want to do, fine by me. Could also use string constants instead of enum, I don't think the numeric values are significant to the implementation but I didn't look at every file on this PR. Eg:

```csharp
public static class MetricReaderTemporalityPreference
{
   public const string AlwaysCumulative = "Cumulative";
   public const string DeltaMonotonicCumulativeNonmonotonic = "Delta";
}
```

Reply:
+1 on being explicit about naming.

BTW, which one is more idiomatic in .NET `OpenTelemetry.Metrics.MetricReaderTemporalityPreference` or `OpenTelemetry.Metrics.MetricReader.TemporalityPreference` (inner class)?